### PR TITLE
[CHI-389] Added Disable support to date picker CE, added Backstop and Cypress tests

### PR DESCRIPTION
**Pull request by janibolkvadze**
@dani-cl-madrid @SergioQCL @mattnickles @jllr

### CODE REVIEW

**Comment by allyourbotsarebelongtous** [comment]
Build FAILED. See http://jenkins.elasticbox.com/job/chi_pull_request/335/ for more details.

**Comment by allyourbotsarebelongtous** [comment]
Build finished SUCCESSFULLY. See http://jenkins.elasticbox.com/job/chi_pull_request/336/ for more details.

File: src/chi/components/date-picker/date-picker.scss [hunk 1]
```diff
@@ -20,6 +20,16 @@ $mondayStartingWeek: (
   sun: 6
 );
 
+.a-inputWrapper.-disabled {
+  pointer-events: none;
+  & .a-input {
```
Comment (SergioQCL) [inline src/chi/components/date-picker/date-picker.scss]: Could these be joined just like this? `& .a-input, & .a-icon {`

File: src/custom-elements/src/components/date-picker/date-picker.tsx [hunk 2]
```diff
@@ -140,10 +145,32 @@ export class DatePicker {
...
+      </chi-popover>
+    );
+
     return [
       // TODO: This input should be chi-input in the future and will pass through
       // some of its configuration attributes. Also will have an icon.
-      <div class="a-inputWrapper -icon--right">
+      <div
+        class={`${
+          this.disabled ? '-disabled' : ''
```
Comment (SergioQCL) [inline src/custom-elements/src/components/date-picker/date-picker.tsx]:
```suggestion
          this.disabled && '-disabled'
```

File: src/custom-elements/src/components/date-picker/date-picker.tsx [hunk 3]
```diff
@@ -155,27 +182,14 @@ export class DatePicker {
...
-        >
-          <chi-date
-            min={this.min}
-            max={this.max}
-            locale={this.locale}
-            value={this.value}
-            format={this.format}
-          />
-        </chi-popover>
+        {!this.disabled ? chiPopover : ''}
```
Comment (SergioQCL) [inline src/custom-elements/src/components/date-picker/date-picker.tsx]:
```suggestion
        {!this.disabled && chiPopover}
```

File: src/website/views/custom-elements/date-picker.pug [hunk 4]
```diff
@@ -49,6 +49,14 @@ p.-text
   :code(lang="html",style="height:20rem")
     <chi-date-picker></chi-date-picker>
 
+h4 Disabled
+.example.-mb--3
+  .-p--3
+    div(style="max-width: 14rem;")
+      chi-date-picker(disabled="true")
+  :code(lang="html",style="height:20rem")
+    <chi-date-picker disabled="true"></chi-date-picker>
```
Comment (SergioQCL) [inline src/website/views/custom-elements/date-picker.pug]:
```suggestion
    <chi-date-picker disabled></chi-date-picker>
```

File: src/website/views/custom-elements/date-picker.pug [hunk 5]
```diff
@@ -49,6 +49,14 @@ p.-text
   :code(lang="html",style="height:20rem")
     <chi-date-picker></chi-date-picker>
 
+h4 Disabled
+.example.-mb--3
+  .-p--3
+    div(style="max-width: 14rem;")
+      chi-date-picker(disabled="true")
```
Comment (SergioQCL) [inline src/website/views/custom-elements/date-picker.pug]: Question: is _true_ value explicitly needed here?

File: test/chi/custom-elements/date-picker.pug [hunk 6]
```diff
@@ -26,3 +26,8 @@ scripts:
   span.a-h2 Input Popover Combined w/ icon
   .-position--relative(style="height: 18rem; width:14rem;")
     chi-date-picker(value=date2, min=date2, active)
+
+.test-disabled(data-cy="test-disabled")
+  span.a-h2 Disabled date picker
+  .-position--relative(style="width:14rem;")
+    chi-date-picker(disabled="true")
```
Comment (SergioQCL) [inline test/chi/custom-elements/date-picker.pug]: Question: is true value explicitly needed here?

Comment (janibolkvadze) [reply test/chi/custom-elements/date-picker.pug]: I tried  `chi-date-picker(disabled)` and it works fine.
All done, thanks for the suggestions!

**Comment by allyourbotsarebelongtous** [comment]
Build finished SUCCESSFULLY. See http://jenkins.elasticbox.com/job/chi_pull_request/337/ for more details.

File: src/chi/components/date-picker/date-picker.scss [hunk 7]
```diff
@@ -20,6 +20,13 @@ $mondayStartingWeek: (
   sun: 6
 );
 
+.a-inputWrapper.-disabled {
+  pointer-events: none;
+  & .a-input, & .a-icon {
```
Comment (jllr) [inline src/chi/components/date-picker/date-picker.scss]: @janibolkvadze .a-input and .a-icon should follow the look and feel of disabled inputs. Please remove this property.

**Pull request merged**